iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/ReplaceThreeWindingsTransformersBy3TwoWindingsTransformers.java

```diff
@@ -0,0 +1,344 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.modification;
+
+import com.powsybl.commons.extensions.Extension;
+import com.powsybl.commons.report.ReportNode;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.modification.topology.NamingStrategy;
+import com.powsybl.iidm.modification.util.ControlledRegulatingTerminals;
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.network.extensions.*;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.powsybl.iidm.modification.util.ModificationReports.*;
+import static com.powsybl.iidm.modification.util.TransformerUtils.*;
+import static com.powsybl.iidm.modification.util.TransformerUtils.copyAndAddPhaseAngleClock;
+
+/**
+ * @author Luma Zamarreño {@literal <zamarrenolm at aia.es>}
+ * @author José Antonio Marqués {@literal <marquesja at aia.es>}
+ */
+public class ReplaceThreeWindingsTransformersBy3TwoWindingsTransformers extends AbstractNetworkModification {
+
+    private static final String THREE_WINDINGS_TRANSFORMER = "ThreeWindingsTransformer";
+
+    @Override
+    public String getName() {
+        return "ReplaceThreeWindingsTransformersBy3TwoWindingsTransformers";
+    }
+
+    @Override
+    public void apply(Network network, NamingStrategy namingStrategy, boolean throwException, ComputationManager computationManager, ReportNode reportNode) {
+        ControlledRegulatingTerminals controlledRegulatingTerminals = new ControlledRegulatingTerminals(network);
+        network.getThreeWindingsTransformerStream().toList()
```

Comment:
Done

Reply:
Sorry, but toList is required for creating a temporary list since the threeWindingsTransformer is removed during the replacement. I undid the last change and added a comment.